rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java

```diff
@@ -686,6 +695,28 @@ public BlockResult getBlockResult(Block b, boolean fullTx) {
         return br;
     }
 
+    public BlockInformationResult[] eth_getBlocksByNumber(String number) {
+        long blockNumber;
+
+        try {
+            blockNumber = StringHexToBigInteger(number).longValue();
+        } catch (NumberFormatException | StringIndexOutOfBoundsException e) {
+            throw new JsonRpcInvalidParamException("invalid blocknumber " + number);
+        }
+
+        List<BlockInformationResult> result = new ArrayList<>();
+        Blockchain blockchain = this.worldManager.getBlockchain();
+
+        synchronized (blockchain) {
```

Comment:
Do not sync over the blockchain object. Check accessLock object in blockchainImpl